src/main/kotlin/cacophony/controlflow/FunctionHandler.kt

```diff
@@ -232,6 +237,118 @@ class FunctionHandlerImpl(
     }
 }
 
+fun defaultCallConvention(index: Int): CFGNode {
+    if (index < REGISTER_ARGUMENT_ORDER.size)
+        return RegisterUse(Register.FixedRegister(REGISTER_ARGUMENT_ORDER[index]))
+    val over = index - REGISTER_ARGUMENT_ORDER.size
+    // Assumes that the RSP has not changed after `call f`
+    return MemoryAccess(
+        Addition(
+            RegisterUse(Register.FixedRegister(HardwareRegister.RSP)),
+            Constant(
+                (over + 1) * REGISTER_SIZE,
+            ),
+        ),
+    )
+}
+
+class PrologueEpilogueHandler(
+    private val handler: FunctionHandlerImpl,
+    private val callConvention: (Int) -> CFGNode,
+    private val preservedRegisters: List<HardwareRegister>, // without RSP
+) {
+    private val numberOfStackVariables =
+        with(handler) {
+            val variables =
+                analyzedFunction
+                    .declaredVariables()
+                    .map { getVariableFromDefinition(it.declaration) } +
+                    analyzedFunction.auxVariables
+            val stackVariables =
+                variables
+                    .map { getVariableAllocation(it) }
+                    .filterIsInstance<VariableAllocation.OnStack>()
+                    .sortedBy { it.offset }
+            run {
+                // Sanity checks
+                var offset = 0
+                for (variable in stackVariables) {
+                    if (variable.offset != offset)
+                        throw CompileException("Holes in stack")
```

Comment:
CompileException should be used for reporting errors in compiled programs, but this on the first glance looks like an error in a compiler?

Reply:
Right, got confused

Reply:
Fixed, but found something similar in the `generateFunctionHandlers` function, not sure if this is a similar issue